homeassistant/components/voice_assistant/__init__.py

```diff
@@ -1,16 +1,90 @@
 """The Voice Assistant integration."""
 from __future__ import annotations
 
-from homeassistant.core import HomeAssistant
+from collections.abc import AsyncIterable
+from typing import Any
+
+from homeassistant.components import stt
+from homeassistant.core import Context, HomeAssistant
 from homeassistant.helpers.typing import ConfigType
 
 from .const import DOMAIN
+from .error import PipelineNotFound
+from .pipeline import (
+    PipelineEvent,
+    PipelineEventCallback,
+    PipelineEventType,
+    PipelineInput,
+    PipelineRun,
+    async_get_pipeline,
+)
 from .websocket_api import async_register_websocket_api
 
+__all__ = (
+    "DOMAIN",
+    "async_setup",
+    "async_pipeline_from_audio_stream",
+    "PipelineEvent",
+    "PipelineEventType",
+)
+
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up Voice Assistant integration."""
     hass.data[DOMAIN] = {}
     async_register_websocket_api(hass)
 
     return True
+
+
+async def async_pipeline_from_audio_stream(
+    hass: HomeAssistant,
+    event_callback: PipelineEventCallback,
+    stt_metadata: stt.SpeechMetadata,
+    stt_stream: AsyncIterable[bytes],
+    language: str | None = None,
+    pipeline_id: str | None = None,
+    conversation_id: str | None = None,
+    context: Context | None = None,
+) -> None:
+    """Create an audio pipeline from an audio stream."""
+    if language is None:
+        language = hass.config.language
+
+    # Temporary workaround for language codes
+    if language == "en":
+        language = "en-US"
```

Comment:
Can we get the user's configured language for cloud STT? It's not guaranteed to match `hass.config.language`, though.

Reply:
this should not be addressed in this PR . This is a copy/paste of how the websocket works.